### PR TITLE
Fix/sortbar hidden when no items

### PR DESCRIPTION
**Pull request by pablocarmu**
## Description

This PR fixes the bug with the sort bar when there were no items inside a folder.

## Motivation and Context

When there are no items inside a folder, the sort bar was hidden by the "No items" view. So you can't tap on the create folder button.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

_____

BUGS & IMPROVEMENTS

- [ ] Default name of a folder https://github.com/owncloud/ios-app/pull/107#issuecomment-415337603

### CODE REVIEW

**Comment by jesmrec** [comment]
### (1)

Suggestion: as hint for the new folder, i would set `New folder`(as web UI) better than `Unknown folder`

File: ownCloud/Client/ClientQueryViewController.swift [hunk 1]
```diff
@@ -19,9 +19,13 @@
 import UIKit
 import ownCloudSDK
 
-class ClientQueryViewController: UITableViewController, Themeable {
-	var core : OCCore?
-	var query : OCQuery?
+class ClientQueryViewController: UIViewController {
```
Comment (felix-schwarz) [inline ownCloud/Client/ClientQueryViewController.swift]: Why did you change the superclass from `UITableViewController` to `UIViewController` and set up the tableView yourself?

File: ownCloud/Client/ClientQueryViewController.swift [hunk 2]
```diff
@@ -123,20 +132,46 @@ class ClientQueryViewController: UITableViewController, Themeable {
...
+			tableView.topAnchor.constraint(equalTo: sortBar.bottomAnchor),
+			tableView.leftAnchor.constraint(equalTo: view.leftAnchor),
+			tableView.rightAnchor.constraint(equalTo: view.rightAnchor),
+			tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+			])
+
+		tableView.delegate = self
+		tableView.dataSource = self
+
+		self.navigationController?.navigationBar.setValue(true, forKey: "hidesShadow")
```
Comment (felix-schwarz) [inline ownCloud/Client/ClientQueryViewController.swift]: This is private API and shouldn't be used.

What effect does it have / why is it needed?

File: ownCloud/Client/ClientQueryViewController.swift [hunk 3]
```diff
@@ -550,18 +590,43 @@ extension ClientQueryViewController : OCQueryDelegate {
 // MARK: - SortBar Delegate
 extension ClientQueryViewController : SortBarDelegate {
 	func sortBar(_ sortBar: SortBar, leftButtonPressed: UIButton) {
-		print("LOG ---> left button pressed")
+		let createFolderVC = NamingViewController(with: core, defaultName: "Unknown Folder".localized, stringValidator: { name in
```
Comment (felix-schwarz) [inline ownCloud/Client/ClientQueryViewController.swift]: Like @jesmrec I think that "New folder" is what a user would usually expect here.

File: ownCloud/Client/SortBar.swift [hunk 4]
```diff
@@ -97,7 +97,7 @@ class SortBar: UIView, Themeable {
 	func applyThemeCollection(theme: Theme, collection: ThemeCollection, event: ThemeEvent) {
 		self.sortButton.applyThemeCollection(collection)
 		self.sortSegmentedControl.applyThemeCollection(collection)
-		self.applyThemeCollection(collection)
+		self.backgroundColor = Theme.shared.activeCollection.navigationBarColors.backgroundColor
```
Comment (felix-schwarz) [inline ownCloud/Client/SortBar.swift]: Please use `collection` instead of `Theme.shared.activeCollection` here.

File: ownCloud/Client/ClientQueryViewController.swift (hunk 1 above)
Comment (pablocarmu) [reply ownCloud/Client/ClientQueryViewController.swift]: The problem this PR solves is that when you don't have any files, the sort bar was hidden by the _No Files_ view. 

There were two approaches:

-  Make the _No Files_ view top constraint constant to be 40 pt (the height of the sort bar) to not hide it with this view. Then the scrolling effect to reload scrolls the sort bar under the _No files_ view and the effect was not beauty.

- Make the `ClientQueryViewController` be a child of `UIViewController` and make the sort bar part of the view controller view.
 The only downside of this is that you need to create the `tableView`  yourself and implement the `UITableViewDelegate` and `UITableViewDataSource`. This way you can have the sort bar independent of the `tableView` and present the _No Files_ view above only the `tableView` allowing the user to push the _create folder_ button and not making any rare scroll with the sort bar.

File: ownCloud/Client/ClientQueryViewController.swift (hunk 2 above)
Comment (pablocarmu) [reply ownCloud/Client/ClientQueryViewController.swift]: The effect was to hide the line between the `NavigationBar` and the sortBar. It's only a visual effect so if this API shouldn't be used we can remove it without any problem.

File: ownCloud/Client/SortBar.swift (hunk 4 above)
Comment (pablocarmu) [reply ownCloud/Client/SortBar.swift]: I'll change it.

File: ownCloud/Client/ClientQueryViewController.swift (hunk 3 above)
Comment (pablocarmu) [reply ownCloud/Client/ClientQueryViewController.swift]: Definitely right. 
I just copied here what files app does but _New Folder_ is a lot better. I'll change it.

**Comment by jesmrec** [comment]
works for me. merging.
